javascript/index.js

```diff
@@ -4,37 +4,175 @@
+                      },
+                      (error) => console.log(error)
+                    );
+                  },
+                  (error) => console.log(error)
+                );
+              },
+              (error) => console.log(error)
+            );
+          },
+          (error) => console.log(error)
+        );
+      },
+      (error) => console.log(error)
+    );
+  },
+  (error) => console.log(error)
+);
 
 // Iteration 2 - using promises
-// ...
+obtainInstruction("steak", 0)
+  .then((step0) => {
+    document.querySelector("#steak").innerHTML += `<li>${step0}</li>`;
+    return obtainInstruction("steak", 1);
+  })
+  .then((step1) => {
+    document.querySelector("#steak").innerHTML += `<li>${step1}</li>`;
+    return obtainInstruction("steak", 2);
+  })
+  .then((step2) => {
+    document.querySelector("#steak").innerHTML += `<li>${step2}</li>`;
+    return obtainInstruction("steak", 3);
+  })
+  .then((step3) => {
+    document.querySelector("#steak").innerHTML += `<li>${step3}</li>`;
+    return obtainInstruction("steak", 4);
+  })
+  .then((step4) => {
+    document.querySelector("#steak").innerHTML += `<li>${step4}</li>`;
+    return obtainInstruction("steak", 5);
+  })
+  .then((step5) => {
+    document.querySelector("#steak").innerHTML += `<li>${step5}</li>`;
+    return obtainInstruction("steak", 6);
+  })
+  .then((step6) => {
+    document.querySelector("#steak").innerHTML += `<li>${step6}</li>`;
+    return obtainInstruction("steak", 7);
+  })
+  .then((step7) => {
+    document.querySelector("#steak").innerHTML += `<li>${step7}</li>`;
+    return obtainInstruction("steak", 8);
+  })
+  .then(() => {
+    document.querySelector(
+      "#steak"
+    ).innerHTML += `<li>${"Steak is ready!"}</li>`;
+  });
+document.querySelector("#steakImg").removeAttribute("hidden");
 
 // Iteration 3 using async/await
-// ...
+async function makeBroccoli() {
+  const item1 = await obtainInstruction("broccoli", 0);
+  document.querySelector("#broccoli").innerHTML += `<li>${item1}</li>`;
+  const item2 = await obtainInstruction("broccoli", 1);
+  document.querySelector("#broccoli").innerHTML += `<li>${item2}</li>`;
+  const item3 = await obtainInstruction("broccoli", 2);
+  document.querySelector("#broccoli").innerHTML += `<li>${item3}</li>`;
+  const item4 = await obtainInstruction("broccoli", 3);
+  document.querySelector("#broccoli").innerHTML += `<li>${item4}</li>`;
+  const item5 = await obtainInstruction("broccoli", 4);
+  document.querySelector("#broccoli").innerHTML += `<li>${item5}</li>`;
+  const item6 = await obtainInstruction("broccoli", 5);
+  document.querySelector("#broccoli").innerHTML += `<li>${item6}</li>`;
+  document.querySelector(
+    "#broccoli"
+  ).innerHTML += `<li>${"Brussels sprouts are ready!"}</li>`;
+  document.querySelector("#broccoliImg").removeAttribute("hidden");
+}
+makeBroccoli();
 
 // Bonus 2 - Promise all
-// ...
\ No newline at end of file
+
+Promise.all(brusselsSprouts)
```

Comment:
this one is wrong. Please compare with the solution in Student Portal.

But all in all great job, Marija!